extralit-server/src/extralit_server/jobs/ocr_jobs.py

```diff
@@ -12,3 +12,243 @@
+from rq.decorators import job
+
+from extralit_server.contexts.ocr.figures import extract_figure_bboxes
+from extralit_server.contexts.ocr.tables import extract_table_bboxes
+from extralit_server.contexts.ocr.text import extract_text_bboxes
+from extralit_server.jobs.queues import DEFAULT_QUEUE, REDIS_CONNECTION
+
+_LOGGER = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+    from marker.renderers.json import JSONOutput
+    from marker.renderers.markdown import MarkdownOutput
+
+try:
+    from marker.config.parser import ConfigParser
+    from marker.converters.pdf import PdfConverter
+    from marker.models import create_model_dict
+except ImportError as e:
+    _LOGGER.error(f"Marker dependencies not available: {e}")
+    raise ImportError("Marker not installed. Install with: pip install marker-pdf") from e
+
+
+@job(queue=DEFAULT_QUEUE, connection=REDIS_CONNECTION, timeout=1800, retry=Retry(max=2, interval=[30, 60]))
+async def async_marker_layout_job(
+    pdf_path: Union[str, Path],
+    pages: Optional[str] = None,
+    extract_text: bool = False,
+    document_id: Optional[UUID] = None,
+) -> dict[str, Any]:
+    """
+    Use Marker to extract layout (tables, figures, text blocks) without running OCR.
+
+    This job uses Marker's layout detection capabilities to identify and extract
+    bounding boxes for different document elements without performing OCR.
+
+    Args:
+        pdf_path: Path to the PDF file to process
+        pages: Optional comma-separated page numbers to process (0-indexed). If None, processes all pages
+        extract_text: Whether to extract text blocks in addition to tables/figures
+        document_id: Optional document ID for job tracking
+
+    Returns:
+        Dictionary containing structured layout information:
+        - tables: List of table bounding boxes
+        - figures: List of figure bounding boxes
+        - text_blocks: List of text block bounding boxes (if extract_text=True)
+        - metadata: Job execution metadata
+    """
+    current_job = get_current_job()
+    if current_job is not None:
+        current_job.meta.update(
+            {
+                "pdf_path": str(pdf_path),
+                "document_id": str(document_id) if document_id else None,
+                "pages": pages,
+                "extract_text": extract_text,
+                "workflow_step": "marker_layout_extraction",
+            }
+        )
+        current_job.save_meta()
+
+    try:
+        pdf_path = Path(pdf_path)
+        if not pdf_path.exists():
+            raise FileNotFoundError(f"PDF file not found: {pdf_path}")
+
+        _LOGGER.info(f"Starting Marker layout extraction for: {pdf_path}")
+
+        # Call Marker's layout detection
+        # Use the real PdfConverter API for layout detection
+        try:
+            # Process the PDF and get layout information
+            layout_result = _call_marker_layout_detection(str(pdf_path), pages)
+        except Exception as e:
```

Comment:
Currently, passing in the `pdf_path` is fine for now as it makes it easier to test/develop, but later on it'll need to be converted to using `s3_url` to download the pdf as bytes, then pass it to `marker` as bytes and not a local file path